packages/api/src/incentives/trading-volume/processSwapEventTradingVolume.ts

```diff
@@ -69,7 +69,12 @@ const processTransactionGroup = (
 ): TradingVolumeItem | null => {
   if (!events.length) return null;
 
-  const accountAddress = highestFeePayerMap.get(transactionId);
+  // For Surge events, use the account address from the event (trading account)
+  // For other events, use the highest fee payer
+  const surgeEvent = events.find((event) => event.accountAddress);
```

Comment:
this is getting the first event that has a `accountAddress`. is this correct in the case of multiple surge events in one batch?